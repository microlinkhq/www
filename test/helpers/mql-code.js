import { test, expect } from 'vitest'

import mqlCode from '../../src/helpers/mql-code'

test('JavaScript with inline code', () => {
  const output = mqlCode('https://microlink.io', {
    function: "({ page }) => page.evaluate('jQuery.fn.jquery')",
    number: 3000,
    boolean: true,
    string: 'MyApiToken',
    url: 'https://myproxy:603f60f5@superproxy.cool:8001',
    scripts: ['https://code.jquery.com/jquery-3.5.0.min.js'],
    multipleScripts: [
      'https://code.jquery.com/jquery-3.5.0.min.js',
      'https://code.jquery.com/jquery-3.5.0.min.js'
    ]
  })

  expect(output.JavaScript()).toMatchSnapshot()
})
