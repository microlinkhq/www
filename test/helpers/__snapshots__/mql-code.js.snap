// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JavaScript with inline code 1`] = `
"
const mql = require('@microlink/mql')

const { status, data } = await mql('https://myproxy:603f60f5@superproxy.cool:8001', { function: '({ page }) => page.evaluate('jQuery.fn.jquery')', number: 3000, boolean: true, string: 'MyApiToken', url: 'https://myproxy:603f60f5@superproxy.cool:8001', scripts: ['https://code.jquery.com/jquery-3.5.0.min.js'], multipleScripts: ['https://code.jquery.com/jquery-3.5.0.min.js','https://code.jquery.com/jquery-3.5.0.min.js'] })

mql.render(data)"
`;
